data_structures/linked_list_find_middle.cpp

```diff
@@ -0,0 +1,149 @@
+    struct Node* slowptr = *head;
+    struct Node* fastptr = *head;
+    //ensure loop runs until fastptr reaches the end of the linked list
+    //when fastptr reaches the end of the list than slowptr will be at
+    //the middle element of the linked list
+    while(fastptr && fastptr->next && fastptr->next->next)
+    {
+        //move by 1 for slowptr
+        slowptr = slowptr->next;
+        //move by 2 for fastptr
+        fastptr = fastptr->next->next;
+    }
+    //store value of middle node and return the integer value
+    int ans = slowptr->val;
+    return ans;
+}
+
+/**
+* @brief Prints the linked list from the inputted array
+* @param int *n array for input
+* @param numNodes number of nodes in linked list
+* @returns void, prints out linked list
+*/
+void printArr(int *n, int numNodes)
+{
+    //takes the array from the main function per user specification
+    int sizeOfArr = numNodes;
+    //prints until the last element of the array has been reached
+    for(int i = 0; i < sizeOfArr; i++)
+    {
+        if(i != sizeOfArr - 1)
+            std::cout<< n[i] << ", ";
+        if(i == sizeOfArr - 1)
+            std::cout<< n[i] << "" << std::endl;
+    }
+}
+
+/**
+* @brief Tests to make sure that all functions work to create a linked list and find the middle element (odd)
+* @returns void
+*/
+static void test1()
+{
+    struct Node* front = nullptr;
+    addNode(&front, 1);
+    addNode(&front, 2);
+    addNode(&front, 3);
+    addNode(&front, 4);
+    addNode(&front, 5);
+    int middle = midPoint(&front);
+    assert (middle == 3);   //ensures Tortoise and Hare algorithm finds the middle element
+}
+
+/**
+* @brief Tests to make sure that all functions work to create a linked list and find the middle element (even)
+* @returns void
+*/
+static void test2()
+{
+    struct Node* front = nullptr;
+    addNode(&front, 1);
+    addNode(&front, 2);
+    addNode(&front, 3);
+    addNode(&front, 4);
+    int middle = midPoint(&front);
+    assert (middle == 3);   //ensures Tortoise and Hare algorithm selects second element in pair of middle elements
+}
+
+/**
+* @brief Main function gathers user input to create a linked list, returning middle element
+* @returns 0 on exit
+*/
+int main()
+{
+    test1();
+    test2();
+    struct Node* head = nullptr;
+    std::cout<< "How many nodes would you like to insert into the Linked List?" << std::endl;
+    int numNodes;
+    std::cin>> numNodes;
+    int *n = new int[numNodes];
+    std::cout<< "Enter the node values (press 'Enter' after each value): " << std::endl;
+
+    for(int i = 0; i < numNodes; i++)
+    {
+        std::cin>> n[i];
+        addNode(&head, n[i]);
+    }
+    
+    printArr(n, numNodes);
+    //destroys dynamically allocated array
+    delete[] n;
+    int midElement = midPoint(&head);
+    std::cout<<std::endl;
+    std::cout<< "The value of the middle element is: " << midElement << std::endl;
+    return 0;
+}
```

Comment:
Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```